src/models/yamlProjectConfiguration.ts

```diff
@@ -264,61 +282,116 @@ export class YamlProjectConfiguration {
 
     const nameRegex = new RegExp("^[a-zA-Z][-a-zA-Z0-9]*$");
     if (!nameRegex.test(configurationFileContent.name)) {
-      throw new Error("The project name is not valid. It must be [a-zA-Z][-a-zA-Z0-9]*");
-    }
-
-    if (configurationFileContent.options && configurationFileContent.options.nodeRuntime && !(
-      configurationFileContent.options.nodeRuntime === "nodejs12.x" ||
-      configurationFileContent.options.nodeRuntime === "nodejs14.x" ||
-      configurationFileContent.options.nodeRuntime === "nodejs16.x" ||
-      configurationFileContent.options.nodeRuntime === "nodejs18.x"
-    )) {
-      throw new Error("The node version in the genezio.yaml configuration file is not valid. The value must be one of the following: nodejs12.x, nodejs14.x, nodejs16.x or nodejs18.x.");
-    }
-
-    if (!configurationFileContent.sdk) {
-      throw new Error("The sdk property is missing from the configuration file.");
+      throw new Error(
+        "The project name is not valid. It must be [a-zA-Z][-a-zA-Z0-9]*"
+      );
     }
 
-    if (!configurationFileContent.sdk.path) {
+    let sdk: YamlSdkConfiguration | undefined;
+    let classes: YamlClassConfiguration[] = [];
+    if (
+      backendConfigurationRequired === BackendConfigurationRequired.BACKEND_REQUIRED &&
+      configurationFileContent.options &&
+      configurationFileContent.options.nodeRuntime &&
+      !(
+        configurationFileContent.options.nodeRuntime === "nodejs16.x" ||
+        configurationFileContent.options.nodeRuntime === "nodejs18.x"
```

Comment:
In the future, we will change this a few times (once new runtimes are launched and old runtimes are being deprecated).

Please create a list `supportedNodeRuntimes = ["nodejs16.x", "nodejs18.x"]` and check if ` configurationFileContent.options.nodeRuntime` is in the list. I think it's gonna be easier to update it.

Reply:
yep, this is much better. done